File: api/src/scripts/test_twitter.ts
import { TwitterApi } from 'twitter-api-v2';
import * as dotenv from 'dotenv';
import * as path from 'path';

// Cargar .env desde la raíz del proyecto
dotenv.config({ path: path.resolve(__dirname, '../../../.env') });

const client = new TwitterApi({
  appKey: process.env.X_API_KEY || '',
  appSecret: process.env.X_API_SECRET || '',
  accessToken: process.env.X_ACCESS_TOKEN || '',
  accessSecret: process.env.X_ACCESS_SECRET || '',
});

async function testTweet() {
  console.log('--- goalworld X (Twitter) Node.js Test ---');
  
  if (!process.env.X_API_KEY || !process.env.X_ACCESS_TOKEN) {
    console.error('❌ Error: Faltan credenciales en el archivo .env');
    return;
  }

  try {
    const tweetText = '⚽ El motor de goalworld (Node.js) está oficialmente en línea.\n\nEl futuro del fútbol descentralizado comienza aquí. 🏆🚀\n\n#goalworld #Solana #Web3 #WorldCup2026';
    
    console.log(`Enviando tweet: \n"${tweetText}"\n`);
    
    const { data: createdTweet } = await client.v2.tweet(tweetText);
    
    console.log('✅ ¡Tweet enviado con éxito!');
    console.log(`🔗 ID del Tweet: ${createdTweet.id}`);
    console.log('🌐 Mira el perfil oficial: https://x.com/goalworldDotFun');
  } catch (error) {
    console.error('❌ Error al enviar el tweet:', error);
    console.log('\nNota: Asegúrate de que las llaves tengan permisos de "Read and Write" en el Developer Portal de X.');
  }
}

testTweet();

File: scripts/test_twitter.py
import os
import tweepy
from dotenv import load_dotenv

# Cargar variables desde el archivo .env en la raíz del proyecto
load_dotenv(os.path.join(os.path.dirname(__file__), '..', '.env'))

# Obtener credenciales
api_key = os.getenv('X_API_KEY')
api_secret = os.getenv('X_API_SECRET')
access_token = os.getenv('X_ACCESS_TOKEN')
access_secret = os.getenv('X_ACCESS_SECRET')
bearer_token = os.getenv('X_BEARER_TOKEN')

print("--- goalworld X (Twitter) Test Script ---")

if not all([api_key, api_secret, access_token, access_secret, bearer_token]):
    print("❌ Error: Faltan credenciales en el archivo .env")
    exit(1)

# Configuración del Cliente v2
client = tweepy.Client(
    bearer_token=bearer_token,
    consumer_key=api_key,
    consumer_secret=api_secret,
    access_token=access_token,
    access_token_secret=access_secret
)

try:
    # Mensaje de prueba con estética goalworld
    tweet_text = "⚽ El motor de goalworld está oficialmente en línea.\n\nPrepárate para la experiencia de fútbol descentralizada más épica del Mundial 2026. 🏆🚀\n\n#goalworld #Solana #NFT #WorldCup2026"
    
    print(f"Enviando tweet: \n\"{tweet_text}\"\n")
    
    response = client.create_tweet(text=tweet_text)
    
    print(f"✅ ¡Tweet enviado con éxito!")
    print(f"🔗 ID del Tweet: {response.data['id']}")
    print(f"🌐 Mira el perfil oficial: https://x.com/goalworldDotFun")

except Exception as e:
    print(f"❌ Error al enviar el tweet: {e}")
    print("\nNota: Asegúrate de que las llaves tengan permisos de 'Read and Write' en el Developer Portal de X.")
